crates/matrix-sdk-ui/src/room_list_service/room_list.rs

```diff
@@ -158,6 +167,51 @@ impl RoomList {
     }
 }
 
+/// This function remembers the current state of the unfiltered room list, so it
+/// knows where all rooms are. When the receiver is triggered, a Set operation
+/// for the room position is inserted to the stream.
+fn merge_stream_and_receiver(
+    mut raw_current_values: Vector<RoomListEntry>,
+    raw_stream: impl Stream<Item = Vec<VectorDiff<RoomListEntry>>>,
+    mut roominfo_update_recv: broadcast::Receiver<RoomInfoUpdate>,
+) -> impl Stream<Item = Vec<VectorDiff<RoomListEntry>>> {
+    let raw_stream_with_recv = stream! {
+        pin_mut!(raw_stream);
+        loop {
+            select! {
+                biased; // Prefer manual updates for easier test code
```

Comment:
Let the code breathe a bit :-) Also helps noticing important details, like this biased keyword here.

```suggestion
                biased; // Prefer manual updates for easier test code
                
```